scripts/constituents.py

```diff
@@ -100,30 +117,462 @@ def find_variable(self, standard_name=None, source_var=None,
+        constituents for this suite."""
+        return "{}_num_consts".format(self.name)
+
+    def const_name_subname(self):
+        """Return the name of the routine that returns a constituent's
+           given an index"""
+        return "{}_const_name".format(self.name)
+
+    def copy_const_subname(self):
+        """Return the name of the routine that returns a copy of a
+           constituent's metadata given an index"""
+        return "{}_copy_const".format(self.name)
+
+    @staticmethod
+    def constituent_index_name(standard_name):
+        """Return the index name associated with <standard_name>"""
+        return "index_of_{}".format(standard_name)
+
+    @staticmethod
+    def write_constituent_use_statements(cap, suite_list, indent):
+        """Write the suite use statements needed by the constituent
+        initialization routines."""
+        maxmod = max([len(s.module) for s in suite_list])
+        smod = len(CONST_DDT_MOD)
+        maxmod = max(maxmod, smod)
+        use_str = "use {},{} only: {}"
+        spc = ' '*(maxmod - smod)
+        cap.write(use_str.format(CONST_DDT_MOD, spc, CONST_PROP_TYPE), indent)
+        cap.write('! Suite constituent interfaces', indent)
+        for suite in suite_list:
+            const_dict = suite.constituent_dictionary()
+            smod = suite.module
+            spc = ' '*(maxmod - len(smod))
+            fname = const_dict.num_consts_funcname()
+            cap.write(use_str.format(smod, spc, fname), indent)
+            fname = const_dict.const_name_subname()
+            cap.write(use_str.format(smod, spc, fname), indent)
+            fname = const_dict.copy_const_subname()
+            cap.write(use_str.format(smod, spc, fname), indent)
+        # end for
+
+    @staticmethod
+    def write_host_routines(cap, host, reg_funcname, num_const_funcname,
+                            copy_in_funcname, copy_out_funcname, const_obj_name,
+                            const_names_name, const_indices_name,
+                            suite_list, err_vars):
+        """Write out the host model <reg_funcname> routine which will
+        instantiate constituent fields for all the constituents in <suite_list>.
+        <err_vars> is a list of the host model's error variables.
+        Also write out the following routines:
+           <num_const_funcname>: Number of constituents
+           <copy_in_funcname>: Collect constituent fields for host
+           <copy_out_funcname>: Update constituent fields from host
+        Output is written to <cap>.
         """
-        const_var = False
-        if isinstance(var, Var):
-            for prop, _ in var.var_properties():
-                if prop in self.__const_prop_names:
-                    const_var = True
-                    break
-                # end if
-            # end for
+# XXgoldyXX: v need to generalize host model error var type support
+        err_callstr = "errflg=errflg, errmsg=errmsg"
```

Comment:
Can this be done by simply looking for the standard name of these two variables on the host model side and treating it like any other variable? Just always add it to the interfaces for the suites and groups (in case there are no schemes to be called that would implicitly add those variables).

Reply:
Yes, that is the idea but first, we need to decide on a replacement for `errflg` and `errmsg`.